blog/_posts/2019-11-13-affiliations-feature.md

```diff
@@ -0,0 +1,78 @@
+
+**Automated Pipelines: assigning identifiers to specific author-affiliation-bibcode entities**
+
+With the initial set of affiliation strings identified we, along with ADS backend programmer Stephen McDonald, designed an automated pipeline to add canonical affiliation identifiers to our database of 15 million references.  It’s a simple process.  First we normalize all of the strings with all uppercase letters, remove spaces and a subset of punctuation marks not needed for disambiguation, which results in a reduced number of matching strings; often, strings will be identical without punctuation, such as strings with “... U.S.A.” and “... USA”.  We then assign the strings to a dictionary, where the strings themselves are the keys, and the identifiers are the values.  If a normalized incoming string matches a key in the dictionary, we assign it the appropriate aff_id, and the record with augmented affiliation data is sent back to our database of metadata for use in the next reindexing process.
+
+We’ve entirely automated the process, so once the dictionary of affiliation strings and matched identifiers is created, it’s a hands-off component of our metadata processing pipeline.  We update dictionaries about once a month, and occasionally more often if we have a large batch of incoming unmatched affiliation strings.  This process is similar to the first curation step, with the only difference being that we prioritize strings according to how many times they appear.  For example, if we receive a number of papers from collaborations with hundreds or thousands of authors *and* their affiliations don’t exactly match our dictionary, we need to add them.  Early on in this process, we came across strings that had many thousands (and sometimes tens of thousands) of occurrences, but the latest batch of unclassified strings appear fewer than 800 times in our entire database. This maximum frequency drops each time we create a new dictionary.
+
+However, we’re getting close to the limit of what we can do with human classification without lots of effort.  Our most recent pass through our metadata has just over 4.6 million unclassified strings.  There are about 6000 unclassified affiliations that appear 100 times or more in the metadata and assigning them aff_ids would provide a lot of new information.  However, over 95 percent of unclassified affiliations occur fewer than 10 times, and **46 percent of affiliation strings only appear once in our affiliation metadata**.
+
+<div class="text-center">
+        <img class="img-thumbnail" src="{{ site.baseurl }}/blog/images/blog_2019-11-13_matched.png" width="50%" /><img class="img-thumbnail" src="{{ site.baseurl }}/blog/images/blog_2019-11-13_unmatched.png" width="50%"/>
+<em>Left: The vast majority of matched publisher-provided affiliation strings in our database appear at least 10 times. Right: XXX</em>
```

Comment:
The bolded part of the paragraph before doesn't match the figure here. Can you check this?

Reply:
I wrote some text, and I'm pasting it here first.  @csgrant00, is there anything in here you think we could use to clarify the existing text?

---
There are 37.4 million affiliation strings -- every affiliation of every author of every paper -- in the ADS.

Of those 37.4 million, 20.7 million (55%) have assigned affiliations;
16.7 million (45%) do not.

If you instead look at the number of *unique* strings -- combining all
identical affiliation strings into one -- there are 6.1 million
unique affiliation strings.     Of those, 1.1 million (18%) have identifiers,
and 5.0 million (82%) do not.   This sounds like we're far behind, but not
really.

```
All affiliation strings:                Unique affiliation strings:
        with IDs        without IDs             with IDs        without IDs
-------------------------------------   ------------------------------------
100K+:    109222(100%)         0( 0%)   100K+:        1(100%)         0( 0%)
50K+ :    235470(100%)         0( 0%)   50K+ :        3(100%)         0( 0%)
10K+ :   1720463(100%)         0( 0%)   10K+ :       88(100%)         0( 0%)
1000+:   8494230(100%)     35101( 0%)   1000+:     2801( 99%)        17( 1%)
100+ :  14465696( 94%)    974412( 6%)   100+ :    25301( 82%)      5436(18%)
10+  :  18410642( 76%)   5873738(24%)   10+  :   167534( 40%)    253717(60%)
2+   :  20264076( 58%)  14472300(42%)   2+   :   676054( 20%)   2744535(80%)
Total:  20732399( 55%)  16746030(45%)   Total:  1144377( 19%)   5018265(81%)
```

The breakdown of number of strings versus how many times they appear in the
database gives some insight.  If you go by the total number of matched
affiliation strings, we have essentially 100 percent coverage of all strings
occurring 1000 or more times among all author/paper affiliations.  That's
around 8.5 million (or 20 percent) of the 37.4 million affiliation strings
we have in the database.

If you look at the total number of unmatched affiliation strings (16.7
million), there's another important statistic: of that 16.7 million,
10.8 million of those occur fewer than 10 times throughout our holdings.
What this means is that there are a sizable number of unidentified strings,
but individually those strings don't appear very often in the database.